src/methods/analyzing/addRank.ts

```diff
@@ -0,0 +1,68 @@
+import { SimpleDataItem } from "../../types/SimpleData.types.js"
+import sortValues from "./sortValues.js"
+
+export default function addRank(
+    data: SimpleDataItem[],
+    rankTitle: string,
+    rankBy?: string | string[],
+    sort?: true | false,
+    locale?: string | (string | undefined | null | boolean)[],
+    handleTies?: "sequential" | "indexWithTie" | "indexWithoutTie"
+): SimpleDataItem[] {
+    if (!rankBy) {
```

Comment:
First, you need to check if the key storing the values does exist, and you need to ensure that the newKey that will be created to add the rank doesn't exist already.

Reply:
the checking for the key existing will happen in the sortValues function and pass the error up. There is now an error for newKey if it already exists